var/spack/repos/builtin/packages/folly/package.py

```diff
@@ -36,11 +37,12 @@ class Folly(AutotoolsPackage):
     # TODO: folly requires gcc 4.9+ and a version of boost compiled with
     # TODO: C++14 support (but there's no neat way to check that these
     # TODO: constraints are met right now)
-    depends_on('boost')
+    depends_on('boost+context cxxstd=11')
```

Comment:
I am not sure we should fix the `cxxstd` here -- different users may want to depend on different versions of boost.  @trws?

Reply:
We should. Folly documents an absolute minimum of 14 for its boost dependence actually, should probably be 17 given that that's what folly provides but at least 14.

Reply:
You shouldn't fix it, you should add conflicts for incompatible cxx versions

Reply:
That is slightly less wrong I suppose.  Because it's an actual requirement of this library though, I'd say the least bad solution with current tools would be to have cxxstd as a variant on here with only `(14, 17, ...)` as options and propagate the selected version to boost.  Conflicts would prevent a bad one from being selected, but wouldn't by itself provide something appropriate by default.